cache/cache.go

```diff
@@ -215,10 +218,21 @@ func (c *Cache) downloadHTTP(b *ui.Task, checksum string, uri string, cachePath
 	// For HTTP files we download and cache them, then return the cached file.
 	task.Debugf("Downloading %s", uri)
 
-	w, response, err := Download(c.httpClient, uri, downloadPath)
+	w, response, err := Download(c.httpClient, nil, uri, downloadPath)
 	if err != nil {
 		return "", "", errors.WithStack(err)
 	}
+	// Check for potential private github release
+	if response.StatusCode == 404 {
+		if ghi, ok := getGithubReleaseInfo(uri); ok {
+			response.Body.Close()
+			w.Close()
+			w, response, err = downloadGHPrivate(c.httpClient, ghi, downloadPath) //nolint:bodyclose // it _is_ handled below.
```

Comment:
the other nolint style in this file (and also in this PR) is `// nolint:` - i.e. a space after the comment marker. We can argue that that is wrong, but it is consistent. Either way, this or the other in this PR should be fixed (line 363).